### PR TITLE
Iss263 fix comma handling in dbc ENUMs

### PR DESCRIPTION
**Pull request by ebroecker**
should fix issue #263

Creating defines out of a string should not be part of canmatrix, but of dbc (maybe also dbf), I want to fix this with another PR.

This PR only should repair the issue.

### CODE REVIEW

File: src/canmatrix/canmatrix.py [hunk 1]
```diff
@@ -1085,14 +1085,15 @@ def safeConvertStrToInt(inStr):
...
-            tempValues = definition[5:].split(',')
+            tempValues = canmatrix.utils.my_comma_splitter(definition[5:])
             self.values = []
             for value in tempValues:
-                value = value.strip()
-                if value[0] == '"':
-                    value = value[1:]
-                if value[-1] == '"':
-                    value = value[:-1]
+                value = value.replace("vector_leerstring", " ")
```
Comment (altendky) [inline src/canmatrix/canmatrix.py]: Should `"vector_leerstring"` be here?  What is it?

And drop the commented code below.

Comment (ebroecker) [reply src/canmatrix/canmatrix.py]: `vector_leerstring` is candb++ way to set an empty string (found out by @blem14 in #263)
 [`leerstring` is german for `empty string`]

File: src/canmatrix/utils.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        return shlex.split(inLine.strip())
+    else:
+        tempArray = shlex.split(inLine.strip().encode('utf-8'))
+        newArray = []
+        for item in tempArray:
+            newArray.append(item.decode('utf-8'))
+        return newArray
+
+
+def my_comma_splitter(string):
```
[the remark this answers is not in the document]
Comment (ebroecker) [reply src/canmatrix/utils.py]: I'm always open and thankful for better english descriptions/names

File: src/canmatrix/tests/test_dbc.py [hunk 3]
```diff
@@ -25,3 +27,23 @@ def test_long_signal_name_imports():
             assert len(line.split()[1]) <= 32
         if line.strip().startswith("BA_ "):
             assert line.split()[5][1:-2] == long_signal_name
+
+
+def test_enum_with_comma():
+    dbc = io.BytesIO(textwrap.dedent(u'''\
+    BA_DEF_ "example0" ENUM "Val1",",";
+    BA_DEF_ BO_ "example1" ENUM "Val 1","vector_leerstring",""," ","'","(",")","[","]","/","-","|","{","}",";",":","<",">",".","?","!","@","#","$","%","^","&","=","`","~";
```
[the remark this answers is not in the document]
Comment (ebroecker) [reply src/canmatrix/tests/test_dbc.py]: a very elegant version. 
Thanks for teaching such nice tricks.
I'll add a second test.

File: src/canmatrix/canmatrix.py (hunk 1 above)
Comment (ebroecker) [reply src/canmatrix/canmatrix.py]: as written above - this is heavy dbc driven. 
Defines should not parse strings in canmatrix, should be done in dbc.  
In another PR I'm gonna cut this out and merge it in  dbc module.

Comment (altendky) [reply src/canmatrix/canmatrix.py]: Didn't they say it was `""` not `" "`?

Comment (ebroecker) [reply src/canmatrix/canmatrix.py]: I'm not sure, will check this tomorrow

Comment (blem14) [reply src/canmatrix/canmatrix.py]: I am sorry you had to wait for response so long, `vector_leerstring` is representing empty string `""`.
The CANdb++ recognizes the `""` when opening a .dbc file, but upon saving it is replaced with `"vector_leerstring"`.

**Pull request merged**